src/start_up/error.rs

```diff
@@ -15,6 +15,7 @@ custom_error! {pub Error
     StorageError { source: StorageError } = "Storage error",
     Blockchain { source: blockchain::LoadError } = "Error while loading the blockchain state",
     Block0 { source: blockcfg::Block0Error } = "Error in the genesis-block",
+    NodeSecrets { source: secure::NodeSecretFromFileError} = "Error while loading the node's secrets."
```

Comment:
... while this one does not.

Reply:
Yes, they are not displayed the same in the main.rs in case of error. The startup::Error is the one being catch at the main level. If it fails it display the error message and then the source. But does not recurse through every sources (that would make UX difficult IMHO).